src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -364,9 +364,14 @@ def operator(self, arg, var_accesses=None):
             symbol_type=LFRicTypes("NumberOfDofsDataSymbol"),
             interface=self._read_access)
 
-        ncells = LFRicTypes("NumberOfCellsDataSymbol")(
-            "ncell_3d", interface=self._read_access)
-        self._symbol_table.add(ncells)
+        try:
+            # We may already have this argument as we add it for each
+            # operator (TODO #2074).
+            ncells = self._symbol_table.lookup("ncell_3d")
+        except KeyError:
+            ncells = LFRicTypes("NumberOfCellsDataSymbol")(
+                "ncell_3d", interface=self._read_access)
+            self._symbol_table.add(ncells)
```

Comment:
Well, that proved complicated. Essentially, we *need* a new symbol table because we need to create symbols with the correct (argument) interfaces. I suspect ArgOrdering gets away with using the symbol table associated with the invoke Schedule because we aren't using interface information there (but we should). So, for now, I do need the `__init__` but I've cut it down and commented it.